$NetBSD: patch-ag,v 1.5 2013/07/06 22:20:33 wiz Exp $

First chunk: libtasn1-3.1 compat.
Chunk 2: Fix building on systems where ulong is not defined.
Chunk 3+4: Fix build failure due to 64-bit time_t.

--- egg/egg-asn1x.c.orig	2010-10-19 02:11:38.000000000 +0000
+++ egg/egg-asn1x.c
@@ -49,8 +49,6 @@
 
 #include "egg-asn1x.h"
 
-#include <libtasn1.h>
-
 #include <stdlib.h>
 #include <string.h>
 
@@ -1988,8 +1986,8 @@ anode_write_integer_ulong (gulong value,
 	guchar buf[sizeof (gulong)];
 	gint bytes, i, off;
 
-	for (i = 0; i < sizeof (ulong); ++i) {
-		off = sizeof (ulong) - (i + 1);
+	for (i = 0; i < sizeof (gulong); ++i) {
+		off = sizeof (gulong) - (i + 1);
 		buf[i] = (value >> (off * 8)) & 0xFF;
 	}
 
@@ -3746,11 +3744,11 @@ egg_asn1x_destroy (gpointer data)
  * TIME PARSING
  */
 
-glong
+time_t
 egg_asn1x_parse_time_general (const gchar *time, gssize n_time)
 {
 	gboolean ret;
-	glong value;
+	time_t value;
 	struct tm when;
 	gint offset = 0;
 
@@ -3777,11 +3775,11 @@ egg_asn1x_parse_time_general (const gcha
 	return value;
 }
 
-glong
+time_t
 egg_asn1x_parse_time_utc (const gchar *time, gssize n_time)
 {
 	gboolean ret;
-	glong value;
+	time_t value;
 	struct tm when;
 	gint offset = 0;
 
